aFWall/src/main/java/dev/ukanth/ufirewall/util/FingerprintUtil.java

```diff
@@ -0,0 +1,395 @@
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+
+import dev.ukanth.ufirewall.Api;
+import dev.ukanth.ufirewall.R;
+
+import static android.content.Context.FINGERPRINT_SERVICE;
+import static android.content.Context.KEYGUARD_SERVICE;
+
+public class FingerprintUtil {
+
+    // generate key based on pkg name
+    public static String GetKey(Context context){
+        return Api.getCurrentPackage(context) + ":Fingerprint";
+    }
+
+    // safely check if device support fingerprint
+    public static boolean isAndroidSupport(){
+        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M;
+    }
+
+    /*
+    * Dialog
+    **/
+    public static class FingerprintDialog extends Dialog{
+
+        private KeyStore keyStore;
+        // Variable used for storing the key in the Android Keystore container
+        private static String KEY_NAME = "Key will be determined at onCreate Event";
+        private Cipher cipher;
+        TextView errorText;
+
+        boolean isNotFirstWindowFocus = false;
+
+        KeyguardManager keyguardManager;
+        FingerprintManager fingerprintManager;
+
+        FingerprintHandler helper;
+
+        FingerprintManager.CryptoObject cryptoObject;
+
+        // callbacks
+        OnFingerprintFailure failureCallback;
+        OnFingerprintSuccess successCallback;
+
+        public FingerprintDialog(Context context) {
+            super(context);
+
+            // Initializing both Android Keyguard Manager and Fingerprint Manager
+            keyguardManager = (KeyguardManager) getContext().getSystemService(KEYGUARD_SERVICE);
+            fingerprintManager = (FingerprintManager) getContext().getSystemService(FINGERPRINT_SERVICE);
+        }
+
+        @Override
+        public void onCreate(@Nullable Bundle savedInstanceState) {
+            super.onCreate(savedInstanceState);
+
+            setCancelable(false);
+            setContentView(R.layout.fingerprint);
+            setTitle(R.string.fingerprint_required);
+
+            errorText = (TextView) findViewById(R.id.fingerprintErrorText);
+
+            // choose key that depends on [pkg_name]:Fingerprint
+            KEY_NAME = GetKey(getContext());
+
+            setFullscreenDialog();
+        }
+
+        // Full screen
+        void setFullscreenDialog(){
+
+            try{
+
+                WindowManager manager = (WindowManager) getContext().getSystemService(Activity.WINDOW_SERVICE);
+                int width, height;
+
+                if (Build.VERSION.SDK_INT > Build.VERSION_CODES.FROYO) {
```

Comment:
**Andorid SDK** older than **Froyo** provide different APIs to specify width & height for the screen, in order to position the dialog.
But, If you see that support older Froyo is not beneficial for the project we can delete if block and leave else block. What is your point? :)